trieste/acquisition/function/function.py

```diff
@@ -223,12 +232,132 @@ def __call__(self, x: TensorType) -> TensorType:
         return (self._eta - mean) * normal.cdf(self._eta) + variance * normal.prob(self._eta)
 
 
+class log_expected_improvement(AcquisitionFunctionClass):
+    def __init__(self, model: ProbabilisticModel, eta: TensorType):
+        r"""
+        Return the Log Expected Improvement (LogEI) acquisition function for single-objective global
+        optimization. Improvement is with respect to the current "best" observation ``eta``, where
+        an improvement moves towards the objective function's minimum and the expectation is
+        calculated with respect to the ``model`` posterior. LogEI mitigates the issue of
+        Expected Improvement (EI) that it is numerically zero in many regions.
+        Although LogEI has (approximately) same optima as EI, it is much easier to optimze,
+        and empirically shows significantly improved optimization performance.
+        See :cite: `ament2023unexpected` for details.
+
+        :param model: The model of the objective function.
+        :param eta: The "best" observation.
+        :return: The expected improvement function. This function will raise
+            :exc:`ValueError` or :exc:`~tf.errors.InvalidArgumentError` if used with a batch size
+            greater than one.
+        """
+        self._model = model
+        self._eta = tf.Variable(eta)
+
+    def update(self, eta: TensorType) -> None:
+        """Update the acquisition function with a new eta value."""
+        self._eta.assign(eta)
+
+    @tf.function
+    @check_shapes(
+        "x: [N..., 1, D] # This acquisition function only supports batch sizes of one",
+        "return: [N..., L]",
+    )
+    def __call__(self, x: TensorType) -> TensorType:
+        eps = (
+            tf.constant(-1e12, dtype=tf.float64)
+            if x.dtype == tf.float64
+            else tf.constant(-1e6, dtype=tf.float32)
+        )
+        mean, variance = self._model.predict(tf.squeeze(x, -2))
+        sigma = tf.sqrt(tf.clip_by_value(variance, clip_value_min=eps, clip_value_max=x.dtype.max))
```

Comment:
see below my comment and perhaps re-using `ensure_positive` used elsehwere for this purpose?

Reply:
Fixed in https://github.com/secondmind-labs/trieste/pull/904/commits/dee5e41e7fbeae04efe9116bae586d309f8772a0